xarray/plot/plot.py

```diff
@@ -32,7 +32,9 @@ def _ensure_plottable(*args):
     Raise exception if there is anything in args that can't be plotted on
     an axis.
     """
-    plottypes = [np.floating, np.integer, np.timedelta64, np.datetime64]
+    from netCDF4 import datetime as nc4datetime
```

Comment:
Yes, looks like we need a separate check for object dtype arrays (it's okay to loop over element).